src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java

```diff
@@ -144,23 +155,35 @@ public void finishLocalSetup() {
         childContext = checkContext;
     }
 
+    /**
+     * {@inheritDoc} Creates child module.
+     * @noinspection ChainOfInstanceofChecks
+     */
     @Override
     public void setupChild(Configuration childConf)
             throws CheckstyleException {
         final String name = childConf.getName();
         final Object module = moduleFactory.createModule(name);
-        if (!(module instanceof AbstractCheck)) {
+        if (module instanceof AutomaticBean) {
```

Comment:
@romani 
As I understood, `TreeWalker` is the child of the `Checker`. So if any exception will occur inside `TreeWalker.setupChild` it will be caught inside `Checker.setupChild` . 

Is this try-catch necessary? 
It will require some tests to be changed, because new thrown exception differs from old one.

Previous exception message

> cannot initialize module com.puppycrawl.tools.checkstyle.TreeWalker - Cannot set property 'excludedPackages' to 'com.puppycrawl.tools.checkstyle.checks.metrics.inputs.a.' in module com.puppycrawl.tools.checkstyle.checks.metrics.ClassFanOutComplexityCheck

Exception with try/catch inside `TreeWalker.setupChild`

> cannot initialize module com.puppycrawl.tools.checkstyle.checks.metrics.ClassFanOutComplexityCheck - Cannot set property 'excludedPackages' to 'com.puppycrawl.tools.checkstyle.checks.metrics.inputs.a.' in module com.puppycrawl.tools.checkstyle.checks.metrics.ClassFanOutComplexityCheck